### PR TITLE
Plane: tiltrotor: add throttle scaling for vectored yaw

### PR DESCRIPTION
**Pull request by IamPete1**
This adds throttle scaling to vectored yaw tilt-rotors. It uses the same scaling method as tailsitters. There scale range is hard coded to the same as the default range for tailsitters (where we have parameters for min and max). So will reduce gains to half at high throttle and double at low throttle. At hover throttle they remain 1 so no change in behavior. We may want to change the max scale factor to 1 so gains are never increased at low throttle, thus making oscillation unlikely. Only yaw is scaled, the roll input used at high tilt angles is not changed. 

It also now sets the yaw limit flags if either tilt is at its limit, this prevents I wind up. 

https://youtu.be/W760-G-N7_I

### CODE REVIEW

File: ArduPlane/tiltrotor.cpp [hunk 1]
```diff
@@ -562,7 +562,16 @@ void Tiltrotor::vectoring(void)
     } else {
         const float yaw_out = motors->get_yaw()+motors->get_yaw_ff();
         const float roll_out = motors->get_roll()+motors->get_roll_ff();
-        float yaw_range = zero_out;
+        const float yaw_range = zero_out;
+
+        // Scaling yaw with throttle
+        const float throttle = motors->get_throttle_out();
+        const float scale_min = 0.5;
+        const float scale_max = 2.0;
```
Comment (IamPete1) [inline ArduPlane/tiltrotor.cpp:570]: We could change this to 1 for more conservative scaling that would not increase gains at low throttle.

File: ArduPlane/tiltrotor.cpp [hunk 2]
```diff
@@ -572,13 +581,28 @@ void Tiltrotor::vectoring(void)
...
+
+        float left_tilt = base_output + tilt_offset;
+        float right_tilt = base_output - tilt_offset;
+
+        // if tilt offset results in output saturation set yaw limit flag
+        // Set flag on both forward and backward tilt limit
+        // this could result in less yaw output in hover as rear limit is hit first
+        // however in that case more yaw effort results in a net forward thrust
+        if ((left_tilt > 1.0) || (left_tilt < 0.0) ||
+            (right_tilt > 1.0) || (right_tilt < 0.0)) {
```
Comment (IamPete1) [inline ArduPlane/tiltrotor.cpp:594]: Could change this to only set limit flag if both tilts are saturated, that would be a smaller change in behaviour.

File: ArduPlane/tiltrotor.cpp (hunk 1 above)
Comment (Gone4Dirt) [reply ArduPlane/tiltrotor.cpp:570]: It is worth noting that these scaling factors were trailed using a lua script on a vehicle with thrust vectoring for yaw and were found to work well for a range of thottles (climbing and descending yaw manouvers in stabilise) so this scaling range does appear appropriate.

File: ArduPlane/tiltrotor.cpp (hunk 2 above)
Comment (tridge) [reply ArduPlane/tiltrotor.cpp:594]: probably do need to make it both saturated, as some vehicles have very little (or no?) back tilt and rely on pitch

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by IamPete1** [comment]
Rebased, now using const throttle and requires both tilts to saturate.

File: ArduPlane/tiltrotor.cpp (hunk 2 above)
Comment (IamPete1) [reply ArduPlane/tiltrotor.cpp]: done

**Review by tridge — APPROVED** [review]

**Pull request merged**